Core/Models/RandomPositions.hpp

```diff
@@ -0,0 +1,162 @@
+//----------------------------------------------------------------------------------------------------
+// File: RandomPositions.hpp
+// Desc: Dynamics model for creating a continuous trajectory
+//       The trajectory is started everytime SyncSystemTo is called, with the initial position
+//       given by the position of the argument. 
+//----------------------------------------------------------------------------------------------------
+#pragma once
+
+// Project-specific
+#include "DynamicsBase.hpp"
+#include "../Utils/Constants.hpp"
+
+namespace gtfo{
+
+template <unsigned int Dimensions, typename Scalar = double>
+class RandomPositionsModel : public DynamicsBase<Dimensions, Scalar>{
+public:
+    using Base = DynamicsBase<Dimensions, Scalar>;
+    using VectorN = typename Base::VectorN;
+
+    RandomPositionsModel(const Scalar& dt, const Scalar& max_speed, const VectorN& joint_limits_lower, const VectorN& joint_limits_upper)
+        :   Base(VectorN::Zero()),
+            dt_(dt),
+            time_(0.0),
+            duration_(0.0),
+            starting_position_(VectorN::Zero()),
+            first_bend_position_(VectorN::Zero()),
+            second_bend_position_(VectorN::Zero()),
+            goal_position_(VectorN::Zero()),
+            direction_(VectorN::Zero()),
+            max_speed_(max_speed),
+            joint_limits_lower_(joint_limits_lower),
+            joint_limits_upper_(joint_limits_upper)
+    {
+        assert(max_speed_ > 0.0);
+        srand(time(0));
+    }
+
+    // Generate a random goal position
+    void SetGoalPositions(const Eigen::Matrix<bool, Dimensions, 1>& joint_locked, const VectorN& starting_position){
+        // generate random goal position
+        for (int i = 0; i < Dimensions; ++i) {
+            if (joint_locked[i]){
+                goal_position_[i] = starting_position[i];
+            } else {
+            goal_position_[i] = (joint_limits_upper_[i] - joint_limits_lower_[i]) * ((double)rand() / RAND_MAX) + joint_limits_lower_[i];}
+        }
```

Comment:
This can be vectorized:
```
const VectorN half_difference = (joint_limits_upper_ - joint_limits_lower_) / 2;
const VectorN half_sum = (joint_limits_upper_ + joint_limits_lower_) / 2;
goal_position_ = joint_locked.select(starting_position, VectorN::Random().cwiseProduct(half_difference) + half_sum);
```